pkg/scheduler/framework/frameworkutils.go

```diff
@@ -0,0 +1,372 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler/framework/uniquename"
+	"go.goms.io/fleet/pkg/utils/controller"
+)
+
+// classifyBindings categorizes bindings into the following groups:
+//   - active bindings, i.e., bindings that are associated with a normally operating cluster and
+//     have been cleared for processing by the dispatcher; and
+//   - creating bindings, i.e., bindings that have been associated with a normally operating cluster,
+//     but have not yet been cleared for processing by the dispatcher; and
+//   - dangling bindings, i.e., bindings that are associated with a cluster that is no longer in
+//     a normally operating state (the cluster has left the fleet, or is in the state of leaving),
+//     yet has not been marked as deleting by the scheduler; and
+//   - obsolete bindings, i.e., bindings that are no longer associated with the latest scheduling
```

Comment:
Hi Ryan! We check for the obsolete bindings for two purposes: a) the scheduling cycle might be the continuation of a prev. one, e.g., the scheduler has created a few bindings (though not all) for the policy snapshot, but failed right after, consequently in the next cycle we would need to take account of these bindings, and b) the design mentions that when scheduling policy changes, for any cluster that was picked before (obsolete) and is now picked again, we would keep the resource placement so that there would be no interruptions